From bda12fc59abe8edc746847f45287ebb1fbe1f0f5 Mon Sep 17 00:00:00 2001
From: Francesco VIRLINZI <francesco.virlinzi@st.com>
Date: Tue, 19 May 2009 09:22:29 +0200
Subject: [PATCH] stm: pm: stx7200 removed the clk_pm_state

Removed the clk_pm_state implementation in the STx7200 SOC,
 because forgotten in the
 [stm: pm: Removed all the clk_pm_state implementation] patch

Signed-off-by: Francesco Virlinzi <francesco.virlinzi@st.com>
---
 arch/sh/kernel/cpu/sh4/clock-stx7200.c |   40 --------------------------------
 1 files changed, 0 insertions(+), 40 deletions(-)

diff --git a/arch/sh/kernel/cpu/sh4/clock-stx7200.c b/arch/sh/kernel/cpu/sh4/clock-stx7200.c
index af83d1c..3d1a84e 100644
--- a/arch/sh/kernel/cpu/sh4/clock-stx7200.c
+++ b/arch/sh/kernel/cpu/sh4/clock-stx7200.c
@@ -613,46 +613,6 @@ static struct clk new_module_clk = {
 	.ops		= &comms_clk_ops
 };
 
-#ifdef CONFIG_PM
-int clk_pm_state(pm_message_t state)
-{
-	static int prev_state = PM_EVENT_ON;
-	unsigned long tmp;
-	switch (state.event) {
-	case PM_EVENT_ON:
-	if (prev_state == PM_EVENT_FREEZE) {
-	} else {
-		/* Restore the GenB.Pll0 frequency */
-		tmp = readl(CLOCKGENB_BASE_ADDR + CLKB_PWR_CFG);
-		writel(tmp & ~CLKB_PLL0_OFF,
-			CLOCKGENB_BASE_ADDR + CLKB_PWR_CFG);
-		/* Wait PllB lock */
-		while ((readl(CLOCKGENB_BASE_ADDR + CLKB_PLL0_CFG)
-			& CLKB_PLL0_LOCK) != 0);
-		tmp = readl(CLOCKGENB_BASE_ADDR + CLKB_PLL0_CFG);
-		writel(tmp & ~CLKB_PLL0_BYPASS,
-			CLOCKGENB_BASE_ADDR + CLKB_PLL0_CFG);
-		mdelay(10); /* wait for stable signal */
-	}
-	break;
-	case PM_EVENT_SUSPEND:
-		/* Reduce the GenB.Pll0 frequency */
-		tmp = readl(CLOCKGENB_BASE_ADDR + CLKB_PLL0_CFG);
-		writel(tmp | CLKB_PLL0_BYPASS,
-			CLOCKGENB_BASE_ADDR + CLKB_PLL0_CFG);
-
-		tmp = readl(CLOCKGENB_BASE_ADDR + CLKB_PWR_CFG);
-		writel(tmp | CLKB_PLL0_OFF,
-			CLOCKGENB_BASE_ADDR + CLKB_PWR_CFG);
-		break;
-	case PM_EVENT_FREEZE:
-		break;
-	}
-	prev_state = state.event;
-	return 0;
-}
-#endif
-
 int __init clk_init(void)
 {
 	int i, ret = 0;
-- 
1.6.0.6

